docs/modules/ROOT/pages/integrations/database.adoc

```diff
@@ -0,0 +1,114 @@
+
+1. Create a new file under `pkg/database` with the name of the new database type.
+2. Include an
+link:https://go.dev/doc/effective_go#init[init function]
+that:
+
+* Inserts in the
+link:https://github.com/maruiz93/kubearchive/blob/postgresql-version-update/pkg/database/database.go#L27[`RegisteredDBCreators`]
+map an entry keyed with the name of the DB type with a
+link:https://github.com/maruiz93/kubearchive/blob/postgresql-version-update/pkg/database/database.go#L24[`newDBCreatorFunc`]
+implementation as value.
+
+* Inserts in the
+link:https://github.com/kubearchive/kubearchive/blob/main/pkg/database/database.go#L26[`RegisteredDatabases`]
+map an entry keyed with the name of the DB type with a
+link:https://github.com/kubearchive/kubearchive/blob/main/pkg/database/database.go#L23[`newDatabaseFunc`]
+as value.
+
+=== The `newDBCreatorFunc`
+
+This function receives the set of environment variables with the database configuration and returns a
+link:https://github.com/kubearchive/kubearchive/blob/main/pkg/database/facade/creator.go[`DBCreator`]
+implementation.
+
+The recommendation is that the `DBCreator` implementation include the `env` param as parameter
+so the `GetDriverName` and the `GetConnectionString` can access them.
+
+=== The `DBInterface` implementation returned by the `newDatabaseFunc`
+
+[NOTE]
+====
+This section includes the development guidelines expected by the KubeArchive maintainers.
+They can be skipped, but they will be considered in the review process.
+====
+
+The `DBInterface` should be a struct with
+link:https://github.com/kubearchive/kubearchive/blob/main/pkg/database/database.go#L40[`Database`]
+embedded.
+
+Make sure to implement the interfaces that are part of it:
+
+* link:https://github.com/kubearchive/kubearchive/blob/main/pkg/database/facade/deleter.go[DBDeleter]
+* link:https://github.com/kubearchive/kubearchive/blob/main/pkg/database/facade/filter.go[DBFilter]
+* link:https://github.com/kubearchive/kubearchive/blob/main/pkg/database/facade/inserter.go[DBInserter]
+* link:https://github.com/kubearchive/kubearchive/blob/main/pkg/database/facade/selector.go[DBSelector]
+* link:https://github.com/kubearchive/kubearchive/blob/main/pkg/database/facade/sorter.go[DBSorter]
+
+[NOTE]
+====
+All the interfaces work with objects from the
+link:https://pkg.go.dev/github.com/huandu/go-sqlbuilder[sqlbuilder Go library].
+Check out the docs to see how to work with them.
+====
+
+[TIP]
+====
+Some interfaces, like `DBDeleter`, have a default implementation.
+
+Other interfaces, like `DBFilter`, have a partial implementation.
+
+Those implementations can have the functionality that you need.
+Check them before implementing your own.
+====
+
+If the database interaction logic changes from the implemented in the `Database` struct,
+override the implementation of their functions.
+An example could be with a database that doesn't support upsert queries
+and need to do a transaction with a `SELECT` and an `INSERT`.
```

Comment:
```suggestion
If the database interaction logic changes from the implemented in the `Database` struct,
override the implementation of their functions. For example, a database that does not
support upsert queries needs to implement `ResourceInserter` as a series of  `SELECT` and an `INSERT` statements.
```

Reply:
no, currently neither mariadb or postgresql reimplement those methods, they just use the "default"implementation of the `Database` struct as it is embedded there. 
It's not correct as go doesn't support the inheritance, it uses composition, but the outcome there is like both `PostgreSQLDatabase` and `MariaDBDatabase` "inherit" the methods from `Database` and they don't need to override them.